x-pack/plugins/fleet/public/applications/fleet/hooks/use_check_permissions.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+
+import { sendGetPermissionsCheck } from '../../../hooks';
+
+export const useCheckPermissions = () => {
+  const [isPermissionsLoading, setIsPermissionsLoading] = useState<boolean>(false);
+  const [permissionsError, setPermissionsError] = useState<string>();
+
+  useEffect(() => {
```

Comment:
anyway we replace that `useEffect` by a react query `useQuery` hook?

Reply:
done